src/experimental/constructs/error-budget-alarm.ts

```diff
@@ -0,0 +1,207 @@
+
+export interface ErrorBudgetAlarmProps {
+  sloName: string;
+  sloTarget: number;
+  badEvents: IMetric;
+  validEvents: IMetric;
+  snsTopicNameForAlerts: string;
+}
+
+interface BurnRate {
+  speed: "Fast" | "Medium" | "Slow";
+  burnRate: number;
+}
+
+interface BurnRateMonitoring extends BurnRate {
+  longPeriod: Duration;
+  shortPeriod: Duration;
+}
+
+interface MonitorBurnRateForPeriodProps {
+  alarmName: string;
+  burnRateSpeed: BurnRate;
+  period: Duration;
+  errorBudget: number;
+  badEvents: IMetric;
+  validEvents: IMetric;
+}
+
+interface CompositeBurnRateAlarmProps {
+  sloName: string;
+  burnRateMonitoring: BurnRateMonitoring;
+  errorBudget: number;
+  badEvents: IMetric;
+  validEvents: IMetric;
+  snsTopic: ITopic;
+  suppressorAlarm?: CompositeBurnRateAlarm;
+}
+
+/**
+ * Use this construct to create alarms that will notify you when you are at risk of missing your SLO targets.
+ *
+ * These alarms follow the recommendations in the [SRE Workbook](https://sre.google/workbook/alerting-on-slos/) to
+ * implement Multiwindow, Multi-Burn-Rate Alerts.
+ *
+ * The CloudWatch implementation of this strategy creates nine separate CloudWatch alarms. The main alarms that users
+ * need to care about are the three composite alarms. These track different burn rates of error budget consumption:
+ * fast (e.g. complete outages), medium (e.g. moderate error percentage over hours, rather than minutes) and slow (e.g.
+ * a small percentage of errors sustained over a much longer period). Multiple burn rates are used to ensure that
+ * we are able to pick up all scenarios which could threaten the SLO, without interrupting developers with an
+ * unnecessary level of urgency.
+ *
+ * Each composite alarm creates two child alarms; these are used to support multiple windows (or periods in CloudWatch
+ * terminology). A composite alarm is configured to send notifications if both of its child alarms are in an alarm state.
+ * The child alarms themselves have no actions/notifications configured and users should not need to interact with them
+ * directly. Multiple windows are helpful for accuracy and reset time. The long window helps us to avoid sending an
+ * alert in scenarios where no intervention is required, for example a very brief spike in errors that is resolved
+ * automatically (e.g. when AWS replaces a 'bad' EC2 instance). The short window helps to ensure that the alert is reset
+ * (or moved back into OK state, in CloudWatch terminology) promptly once a problem has been resolved.
+ *
+ * The composite alarms have an awareness of priority, meaning that low priority alarm notifications are suppressed if
+ * a higher priority alarm is already firing. That is, if you receive an alert about fast error budget consumption, you
+ * should not receive an alert from the medium or slow versions of the alarm.
+ */
+export class ErrorBudgetAlarmExperimental {
+  // TODO: Should this extend Construct?
```

Comment:
I noticed that a couple of our other classes (which are not subclasses of an AWS construct) extend `Construct` e.g. [here](https://github.com/guardian/cdk/blob/10762e709cbf9686364ece07b1738b8a493a0b0b/src/patterns/ec2-app/base.ts#L331) and [here](https://github.com/guardian/cdk/blob/043953d45c9337935f5c8ab89fbb072905d3068b/src/constructs/dns/dns-records.ts#L24).

Do I need to do something similar? cc @akash1810 @nicl

Reply:
I believe this enables better tracking of GuCDK usage: https://github.com/guardian/cdk/pull/1532

Reply:
Ah, thanks for the link / explanation! I'm not sure that we strictly need this yet but I've added it anyway as it was trivial to change this (and it saves me from forgetting about it later!)

Reply:
There might be a need to signpost this more clearly, either in the contributing docs or in a features page.